backend/src/routes/api/medication.routes.js

```diff
@@ -0,0 +1,52 @@
+import Medication, { medicationValidator } from '../../models/Medication.js';
+import { StatusCodes } from 'http-status-codes';
+import ExpressError from '../../middlewares/ExpressError.js';
+
+const MedicationRoutes = (router) => {
+  router.get('/me/meds', getAllMedications);
+  router.post('/me/meds', medicationValidator, postMedication);
+};
+
+async function getAllMedications(_req, res, next) {
+  try {
+    const medications = await Medication.find({});
+    if (!medications) {
+      throw new ExpressError('No medication found', 404);
+    }
+    res.status(200).send({ medications });
+  } catch (err) {
+    next(err);
+  }
+  next();
+}
+
+async function postMedication(req, res, next) {
+  try {
+    const newMedication = await addNewMedication(req.body);
+    res.status(StatusCodes.CREATED).json(newMedication);
```

Comment:
newMedication will be undefined I think, because we are not returning anything from addNewMedication